### PR TITLE
Corrections to V7 transaction format and using updated test vectors

### PR DESCRIPTION
**Pull request by vivek-arte**
This PR makes the following changes:
 - reordering of the component fields in `read_action` and `write_action` in issuance in order to make them spec compliant. 
 - It sets some of the constant values for NU7 in a way that is consistent with the generated test vectors from the Python reference implementation (See https://github.com/QED-it/zcash-test-vectors/pull/22 for the changes to the Python reference implementation).
 - Some changes are made to function names to reflect the shift from preparing ZSAs for V7 rather than V6.
 - There are changes to the tests to use the `zcash_unstable` flag in order to cover both ZSA and vanilla Orchard behavior.
 - The test vectors are also expanded to include both the V5 vectors, and the V7 vectors (behind a `zcash_unstable` flag).

### CODE REVIEW

File: zcash_primitives/src/transaction/mod.rs [hunk 1]
```diff
@@ -58,7 +58,7 @@ const V5_VERSION_GROUP_ID: u32 = 0x26A7270A;
 #[cfg(zcash_unstable = "nu6")] /* TODO nu7 */
 const V7_TX_VERSION: u32 = 7;
 #[cfg(zcash_unstable = "nu6")] /* TODO nu7 */
-const V7_VERSION_GROUP_ID: u32 = 0x26A7270C; // TODO ???
+const V7_VERSION_GROUP_ID: u32 = 0x124A69F8; // TODO ???
```
Comment (alexeykoren) [inline zcash_primitives/src/transaction/mod.rs:61]: If this is final - please remove TODO comment

**Review by alexeykoren — CHANGES_REQUESTED** [review]
Good stuff, one minor question

File: zcash_primitives/src/transaction/mod.rs (hunk 1 above)
Comment (vivek-arte) [reply zcash_primitives/src/transaction/mod.rs:61]: I picked a value (at random) in `zcash-test-vectors`, and set this to match that value. I'm not sure if there is a value suggested by Zcash yet. That's why I left the TODO comment.

Comment (alexeykoren) [reply zcash_primitives/src/transaction/mod.rs:61]: Ah, ok, then keep it

**Review by alexeykoren — APPROVED** [review]

**Comment by vivek-arte** [comment]
CI checks fail due to missing `NoteBytes` in `zcash_note_encryption`, which seems to be getting resolved in https://github.com/QED-it/zcash_note_encryption/pull/8.

**Comment by alexeykoren** [comment]
> CI checks fail due to missing `NoteBytes` in `zcash_note_encryption`, which seems to be getting resolved in [QED-it/zcash_note_encryption#8](https://github.com/QED-it/zcash_note_encryption/pull/8).

Right. Yes, we'll merge it soon. But this PR we can merge even before, given that we are not targeting zsa1

File: zcash_primitives/src/transaction/tests.rs [hunk 2]
```diff
@@ -236,7 +236,12 @@ fn zip_0244() {
     fn to_test_txdata(
         tv: &self::data::zip_0244::TestVector,
     ) -> (TransactionData<TestUnauthorized>, TxDigests<Blake2bHash>) {
-        let tx = Transaction::read(&tv.tx[..], BranchId::Nu5).unwrap();
+        let tx = Transaction::read(
+            &tv.tx[..],
+            #[cfg(not(zcash_unstable = "nu6"))] /* TODO nu7 */ BranchId::Nu5,
```
Comment (PaulLaux) [inline zcash_primitives/src/transaction/tests.rs:241]: new line after `*/` (all occurrences)

**Review by PaulLaux — APPROVED** [review]
please fix CI error so we can merge it.

File: zcash_primitives/src/transaction/tests.rs (hunk 2 above)
Comment (vivek-arte) [reply zcash_primitives/src/transaction/tests.rs:241]: `cargo fmt` pushes it back to the above line (so making it new line would make it fail the Rustfmt CI check).

Comment (vivek-arte) [reply zcash_primitives/src/transaction/tests.rs:241]: I confirmed that this is the case for all lines where the newline is not used.

**Pull request merged**